device.go

```diff
@@ -78,7 +80,19 @@ func (d *device) InstallUpdate(image io.ReadCloser, size int64) error {
 		return err
 	}
 
-	b := &BlockDevice{Path: inactivePartition}
+	typeUBI := sysfs.Class.Object("ubi").SubObject(inactivePartition).Exists()
```

Comment:
Maybe move this also to `ioctl.go`. It's not as clear a match as the syscall, but sysfs is definitely Linux specific as well.